### PR TITLE
Fix nahnoji.cz

### PR DESCRIPTION
**Pull request by jose1711**
This change is [Reviewable](https://reviewable.io/reviews/kodi-czsk/script.module.stream.resolver/31)

### CODE REVIEW

File: lib/server/nahnojiresolver.py [hunk 1]
```diff
@@ -21,10 +21,10 @@ def url(url):
 		ur = m.group('url')
 		if ur.find('.flv') > -1:
 			return [ur]
-		data = util.substr(util.request(ur),'<body style','</a>')
-		pattern = 'href=\"(.+?)\"[^>]+>' 
+		data = util.substr(util.request(ur),'<source type="video/mp4"','</source>')
+		pattern = 'src=\"(.+?)\"[^>]+>'
```
Comment (lubo) [inline lib/server/nahnojiresolver.py]: What about using `util.parse_html()` ?

Comment (jose1711) [inline lib/server/nahnojiresolver.py]: sadly, this is not an option as html is (purposely?) invalid as beautifulsoup is unable to process it properly. here's an example: `<html><head>..</head></html><body>..</body>`

Comment (lubo) [inline lib/server/nahnojiresolver.py]: I've no problems with BeautifulSoap on html5lib. I use the function mentioned above with multie plugins with great results. You should definitely test it out.

Comment (jose1711) [inline lib/server/nahnojiresolver.py]: I stand corrected - wasn't using 'html5lib' but 'html' parser instead, sorry

File: lib/server/nahnojiresolver.py [hunk 2]
```diff
@@ -15,16 +15,14 @@ def supports(url):
...
 		ur = m.group('url')
 		if ur.find('.flv') > -1:
 			return [ur]
-		data = util.substr(util.request(ur),'<body style','</a>')
-		pattern = 'href=\"(.+?)\"[^>]+>' 
-		match = re.compile(pattern).findall(data)
-		return [match[0]]
+                page = util.parse_html(ur)
+                data = page.select('source[type=video/mp4]')[0]['src']
+		return ['http://nahnoji.cz'+data]
```
Comment (lubo) [inline lib/server/nahnojiresolver.py]: Are you sure that there's always only one stream available ?

File: lib/server/nahnojiresolver.py [hunk 3]
```diff
@@ -15,16 +15,14 @@ def supports(url):
 
 # returns the steam url
 def url(url):
-	#if supports(url):
 	m = _regex(url)
 	if not m == None:
 		ur = m.group('url')
```
Comment (lubo) [inline lib/server/nahnojiresolver.py]: If you rewrite `_regex(url)` correctly, you won't have to do this again.

**Comment by lubo** [comment]
Overall quality of this particular resolver is really poor. It'd be nice to rewrite it.

**Comment by jose1711** [comment]
please check now, i've done some more modifications

File: lib/server/nahnojiresolver.py [hunk 4]
```diff
@@ -15,21 +15,19 @@ def supports(url):
...
+		return ['http://nahnoji.cz'+data]
 
 def resolve(u):
 	stream = url(u)
 	if stream:
 		return [{'name':__name__,'quality':'360p','url':stream[0],'surl':u}]
 
 def _regex(url):
-	return re.search('(?P<url>http://nahnoji.cz/[^\"|\'|\\\]+)',url,re.IGNORECASE | re.DOTALL)
+	return re.search(r'(?P<url>http://nahnoji.cz/((?P<flv>.+\.flv)|(.+)))$',url,re.IGNORECASE | re.DOTALL)
```
Comment (lubo) [inline lib/server/nahnojiresolver.py]: This overly complex regex is not really necessary and not the most performant one. I'd remove this function and use something like `re.search(r'nahnoji\.cz/.+', url) is not None` in `supports(url)` and `re.search(r'\.flv$', url)` for checking for URLs ending with `.flv`.

File: lib/server/nahnojiresolver.py [hunk 5]
```diff
@@ -15,21 +15,19 @@ def supports(url):
...
 		ur = m.group('url')
-		if ur.find('.flv') > -1:
-			return [ur]
-		data = util.substr(util.request(ur),'<body style','</a>')
-		pattern = 'href=\"(.+?)\"[^>]+>' 
-		match = re.compile(pattern).findall(data)
-		return [match[0]]
+		if m.group('flv'): return [ur]
+		page = util.parse_html(ur)
+		data = page.select('source[type=video/mp4]')[0]['src']
```
Comment (lubo) [inline lib/server/nahnojiresolver.py]: Why do you always return the first instance only ? Returning all of them costs nothing. Also, I'd move all the code from this function to `resolve(url)` to make it more simple.

**Pull request merged**